firewood/src/merkle/node.rs

```diff
@@ -310,6 +310,10 @@ impl ExtNode {
         self.1
     }
 
+    pub fn chd_encoded(&self) -> &Option<Vec<u8>> {
+        &self.2
```

Comment:
You should never return references to `Option`. Instead, return `Option<&[u8]>` 
```suggestion
    pub fn chd_encoded(&self) -> Option<&[u8]> {
        self.2.as_ref()
```